### PR TITLE
DROID-74: framework ble package refactor

### PR DESCRIPTION
**Pull request by miwright2**

### CODE REVIEW

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+ * @param adapter Android BluetoothAdapter.
+ */
+internal open class BleDevice (
+    private val mac: String,
+    protected val owner: LifecycleOwner,
+    var advertisingData: LegacyProbeAdvertisingData,
+    adapter: BluetoothAdapter
+) {
+    companion object {
+        private const val PROBE_REMOTE_RSSI_POLL_RATE_MS = 2000L
```
Comment (nick-sasquatch) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt:62]: If these are the same across all devices, I'd suggest removing the `PROBE_` prefix.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                // engine = SystemLogEngine
+                // level = Logging.Level.Events
+                // format = Logging.Format.Multiline
+                // data = Hex
+                 */
+            }
+        }
+
+    protected val connectionMonitor = IdleMonitor()
+    val remoteRssi = AtomicInteger(0)
```
Comment (nick-sasquatch) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt:96]: Spoke with Matt about this--please put a `TODO` in here to make private when advertising data injection's addressed.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+                }
+            }
+        })
+    }
+
+    fun observeRemoteRssi(callback: (suspend (rssi: Int) -> Unit)? = null) {
+        jobManager.addJob(owner.lifecycleScope.launch(Dispatchers.IO) {
+            var exceptionCount = 0;
+            while(isActive) {
+                if(isConnected.get() && mac != SimulatedProbeManager.SIMULATED_MAC) {
```
Comment (nick-sasquatch) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt:126]: Can the simulated probe be handled closer to the probe device implementation?

**Review by nick-sasquatch — APPROVED** [review]
Just a couple minor comments--approving pending comments being addressed. Nice work.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt (hunk 2 above)
Comment (miwright2) [reply combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt:96]: I hope this captures the design intent for when scanning is integrated.

```
    // TODO: This variable should ideally be scoped as private.  
    //  This should be possible when this class is able handle and process advertising
    //  packets directly from the device scanner.
```

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt (hunk 3 above)
Comment (miwright2) [reply combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt:126]: I think so.  I put together the following design notes under DROID-62 that I think would cover that... I'll start a thread in Slack for opinions.

image

**Pull request merged**